### ISSUE TITLE
Zero-length `notify` list causes a TypeError and traceback

### ISSUE THREAD

**Issue by angstwad** [ansible/ansible#11228]
I noticed this after a typo in my playbook.  Running 1.9.1 on OS X 10.10.  Looks like a zero-length `notify` list (or yaml array) causes a type error and traceback because `NoneType` doesn't have a `__len__`.  

I assume you want this to trigger a syntax warning or something rather than a traceback.  

I can probably do a PR if you're interested.

```
Traceback (most recent call last):
  File "/Users/paul4611/Projects/Python/virtualenvs/playbooks/bin/ansible-playbook", line 324, in <module>
    sys.exit(main(sys.argv[1:]))
  File "/Users/paul4611/Projects/Python/virtualenvs/playbooks/bin/ansible-playbook", line 264, in main
    pb.run()
  File "/Users/paul4611/Projects/Python/virtualenvs/playbooks/lib/python2.7/site-packages/ansible/playbook/__init__.py", line 348, in run
    if not self._run_play(play):
  File "/Users/paul4611/Projects/Python/virtualenvs/playbooks/lib/python2.7/site-packages/ansible/playbook/__init__.py", line 789, in _run_play
    if not self._run_task(play, task, False):
  File "/Users/paul4611/Projects/Python/virtualenvs/playbooks/lib/python2.7/site-packages/ansible/playbook/__init__.py", line 549, in _run_task
    if len(task.notify) > 0:
TypeError: object of type 'NoneType' has no len()
```

**Comment by bcoca** [comment]
this seems to have been fixed in current development

**Comment by bcoca** [comment]
firx for 1.9 via 74afd2438754a2c94e37c3bfe2804c43b7f6c7f3, it should be released with the upcomming 1.9.2
